### PR TITLE
Tests for all classes and methods have been written.

### PR DESCRIPTION
**Pull request by sshestaka**
Edge case handling added to methods and tests.

### CODE REVIEW

File: src/test/java/core/basesyntax/FruitShopServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+public class FruitShopServiceTest {
+    private static OperationStrategy operationStrategy;
+    private static List<FruitTransaction> fruitTransactionList;
+    private static Map<String, Integer> fruitsStorageExpected;
+    private static Map<OperationType, OperationHandler> operationTypeMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        FruitTransaction fruitTransaction1 = new FruitTransaction(OperationType.BALANCE,
```
Comment (Ivan95kos) [inline src/test/java/core/basesyntax/FruitShopServiceTest.java:35]: It is better use more informative names
Using numbers in names is bad practice, fix everywhere
```suggestion
        FruitTransaction bananaForBalance = new FruitTransaction(OperationType.BALANCE,
```

File: src/test/java/core/basesyntax/OperationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import core.basesyntax.strategy.impl.ReturnOperationHandlerImpl;
+import core.basesyntax.strategy.impl.SupplyOperationHandlerImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class OperationHandlerTest {
+    //Balance Handle Tests
```
Comment (Ivan95kos) [inline src/test/java/core/basesyntax/OperationHandlerTest.java:21]: please, remove the comments

File: src/test/java/core/basesyntax/OperationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import core.basesyntax.strategy.impl.PurchaseOperationHandlerImpl;
+import core.basesyntax.strategy.impl.ReturnOperationHandlerImpl;
+import core.basesyntax.strategy.impl.SupplyOperationHandlerImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class OperationHandlerTest {
```
Comment (Ivan95kos) [inline src/test/java/core/basesyntax/OperationHandlerTest.java:20]: You should have TestClass for each handler, no general

File: src/test/java/core/basesyntax/FruitTransactionTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Test
+    public void add_nullNameToFruitTransaction_NotOk() {
+        assertThrows(ValidationException.class,
+                () -> {
+                    FruitTransaction nullTransaction =
+                            new FruitTransaction(OperationType.BALANCE,
+                                    null, 10);
+                });
+    }
```
Comment (Ivan95kos) [inline src/test/java/core/basesyntax/FruitTransactionTest.java:51]: This looks better:
```
    @Test
    public void add_nullNameToFruitTransaction_NotOk() {
        assertThrows(ValidationException.class, () -> {
                    new FruitTransaction(OperationType.BALANCE, null, 10);
                });
    }
```

File: src/test/java/core/basesyntax/FruitTransactionTest.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exception.ValidationException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationType;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FruitTransactionTest {
```
Comment (Ivan95kos) [inline src/test/java/core/basesyntax/FruitTransactionTest.java:11]: For the future, you should test only logic in Services, Perositories, Handlers
Test cases are rarely written for models

File: src/test/java/core/basesyntax/FruitShopServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
+package core.basesyntax;
```
Comment (Ivan95kos) [inline src/test/java/core/basesyntax/FruitShopServiceTest.java:1]: The structure for the tests should be the same as the structure of the project with all packages(sevice, model, strategy....)

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Good job overall!

File: src/test/java/core/basesyntax/FruitShopServiceTest.java (hunk 1 above)
Comment (sshestaka) [reply src/test/java/core/basesyntax/FruitShopServiceTest.java:35]: Ok, thanks!

File: src/test/java/core/basesyntax/OperationHandlerTest.java (hunk 2 above)
Comment (sshestaka) [reply src/test/java/core/basesyntax/OperationHandlerTest.java:21]: Oook))

File: src/test/java/core/basesyntax/OperationHandlerTest.java (hunk 3 above)
Comment (sshestaka) [reply src/test/java/core/basesyntax/OperationHandlerTest.java:20]: Ok, I did it.

File: src/test/java/core/basesyntax/FruitTransactionTest.java (hunk 4 above)
Comment (sshestaka) [reply src/test/java/core/basesyntax/FruitTransactionTest.java:51]: Ok) I have replaced

File: src/test/java/core/basesyntax/FruitTransactionTest.java (hunk 5 above)
Comment (sshestaka) [reply src/test/java/core/basesyntax/FruitTransactionTest.java:11]: Thanksm now I know it)

File: src/test/java/core/basesyntax/FruitShopServiceTest.java (hunk 6 above)
Comment (sshestaka) [reply src/test/java/core/basesyntax/FruitShopServiceTest.java:1]: Ok, I created packages for each test.

**Review by aaananas — COMMENTED** [review]
closed, reopen your pr

**Pull request closed without merging**